controller/user_controller.js

```diff
@@ -0,0 +1,180 @@
+const AddressSchema = require("../model/userAddressSchema");
+const ProfileSchema = require("../model/userProfileSchema");
+const UserData = require("../model/userSchema");
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+
+const userRegister = async (req, res) => {
+  try {
+    const {
+      first_name,
+      last_name,
+      user_name,
+      email,
+      password,
+      confirm_password,
+    } = req.body;
+
+    const emailRegexp =
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+    if (emailRegexp.test(email) == false) {
+      throw new Error("email is not valid");
+    }
+
+    if (password !== confirm_password) {
+      throw new Error("password is not matching");
+    }
+    if (!email) {
+      throw new Error("email is required");
+    }
+    if (!first_name) {
+      throw new Error("first_name is required");
+    }
+    if (!password) {
+      throw new Error("password is required");
+    }
+    if (!last_name) {
+      throw new Error("last_name is required");
+    }
+    if (!user_name) {
+      throw new Error("user_name is required");
+    }
+    if (!(email && password && first_name && last_name && user_name)) {
+      throw new Error("All input is required");
+    }
+    const oldUser = await UserData.findOne({ email });
+    if (oldUser) {
+      return res.status(409).send("User Already Exist. Please Login");
+    }
+    encryptedPassword = await bcrypt.hash(password, 10);
+    const user = await UserData.create({
+      first_name,
+      last_name,
+      email: email.toLowerCase(), // sanitize: convert email to lowercase
+      password: encryptedPassword,
+    });
+    const token = jwt.sign(
+      { user_id: user._id, email },
+      process.env.TOKEN_KEY,
+      {
+        expiresIn: "2h",
+      }
+    );
+    user.token = token;
+    const data = await user.save();
+    const response = {
+      first_name: data.first_name,
+      last_name: data.last_name,
+      user_name: data.user_name,
+      email: data.email,
+      token: data.token,
+    };
+    res.status(201).json(response);
+  } catch (err) {
+    res.send(err.message);
+  }
+};
+
+const userLogin = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+    if (!(email && password)) {
+      res.status(400).send("All input is required");
```

Comment:
this condition will only work if both email && password is not supplied , however we need to check both email and password is supplied or not seperately